### PR TITLE
Jetpack: update links on single products

### PR DESCRIPTION
**Pull request by keoshi**
#### Changes proposed in this Pull Request

* Updates copy.
* Removes “Which backup option is best for me?” link from the products header.
* Updates products constants to add link properties.
* Adds links to single product cards.
* Complementary to https://github.com/Automattic/jetpack/pull/15237

#### Testing instructions

* Fire up this PR
* Go to `http://calypso.localhost:3000/plans/SITE_URL`.
* Ensure: 
  * All single product cards look good.
  * Both have links that redirect to their respective LPs.
  * Tracking works.

#### Before
image

#### After

image

Fixes #15183 along with https://github.com/Automattic/jetpack/pull/15237

### CODE REVIEW

File: client/my-sites/plans-features-main/products-header.jsx [hunk 1]
```diff
@@ -4,75 +4,29 @@
...
-					tracksEventProps={ {
-						link_location: 'product_jetpack_backup_description',
-						link_slug: 'which-one-do-i-need',
-					} }
-					icon
-				>
-					{ translate( 'Which backup option is best for me?' ) }
-				</ExternalLinkWithTracking>
-			</Fragment>
+			<Fragment>{ translate( "Looking for specific features? We've got you covered." ) }</Fragment>
```
Comment (a8ci18n) [inline client/my-sites/plans-features-main/products-header.jsx:29]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Just looking for backups? We've got you covered.' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/jetpack-plans-page-single-products-links)

**Comment by rcanepa** [comment]
Hey @keoshi! I tested it locally and works great. 

I noticed a tiny issue that was introduced [here](https://github.com/Automattic/wp-calypso/pull/40621/commits/2a87c0051f2c07f1e9416c0c88ca498192f1d829#diff-67e84bb8c75685ed38b07fe2d8d7dd74R282). That change made the `optionShortNames` contain a string instead of an object and React is complaining about it. 

image

[Here you can see that the `ProductSelector` component](https://github.com/Automattic/wp-calypso/blob/5c7f52a8dbe9d18495ab0fc558a1856033c0cd72/client/blocks/product-selector/index.jsx#L55) expects that prop to be an object.

Sorry for not catching this in the other PR.

**Comment by keoshi** [comment]
@rcanepa Thanks for catching that, I did not notice it.

Should I use this instead `optionShortNames: getJetpackProductsShortNames(),` ?

**Comment by keoshi** [comment]
@rcanepa updated in https://github.com/Automattic/wp-calypso/pull/40672/commits/4de474136c51b9660a20bd0cd67097b358ace1f9 — that seems to fix it, I don't see the error anymore. Can you give it another quick spin and validate, please?

**Comment by rcanepa** [comment]
@keoshi it indeed fixed the problem! It looks fantastic!

**Comment by keoshi** [comment]
@jeffgolenski @eeeeevon13 @crunnells Would appreciate if any of you could take this for a spin. Thank you. 🙌

**Pull request merged**